AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1282,19 +1281,51 @@ - (void)_tearDownPlaceholderLayer
 
 #pragma mark - For Subclasses
 
+- (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  if (_methodOverrides & ASDisplayNodeMethodOverrideCalculateSizeThatFits) {
+    CGSize size = [self calculateSizeThatFits:constrainedSize.max];
+    return [ASLayout newWithLayoutableObject:self size:ASSizeRangeClamp(constrainedSize, size)];
+  } else {
+    id<ASLayoutable> layoutSpec = [self layoutSpecThatFits:constrainedSize];
+    ASLayout *layout = [layoutSpec measureWithSizeRange:constrainedSize];
+    // Make sure layoutableObject of the root layout is `self`, so that the flattened layout will be structurally correct.
+    if (layout.layoutableObject != self) {
+      layout.position = CGPointZero;
+      layout = [ASLayout newWithLayoutableObject:self size:layout.size sublayouts:@[layout]];
+    }
+    return [layout flattenedLayoutUsingPredicateBlock:^BOOL(ASLayout *evaluatedLayout) {
+      return [_subnodes containsObject:evaluatedLayout.layoutableObject];
+    }];
+  }
+}
+
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDisplayNodeAssertThreadAffinity(self);
   return CGSizeZero;
 }
 
+- (id<ASLayoutable>)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  return nil;
```

Comment:
I'm not sure if it would be necessary to enter here, but I think if a root class ASDisplayNode instance responded to a -measure: call by returning a layout with some size in between the min and max.  This would make it easier to manage nodes created to show a .backgroundColor layer or highlighting line.